.pre-commit-config.yaml

```diff
@@ -1,45 +1,66 @@
+# All available hooks: https://pre-commit.com/hooks.html
+# R specific hooks: https://github.com/lorenzwalthert/precommit
 repos:
-#####
-# Basic file cleanliness
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.5.0
+# R
+-   repo: https://github.com/lorenzwalthert/precommit
+    rev: v0.4.2
     hooks:
+    -   id: style-files
+        args: [--style_pkg=styler, --style_fun=tidyverse_style]    
+    -   id: roxygenize
+    -   id: use-tidy-description
+    -   id: lintr
+    -   id: readme-rmd-rendered
+    -   id: parsable-R
+    -   id: no-browser-statement
+    -   id: no-print-statement
+    -   id: no-debug-statement
+    -   id: deps-in-desc
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks: 
     -   id: check-added-large-files
-    -   id: check-yaml
-    -   id: check-toml
+        args: ['--maxkb=200']
+    -   id: file-contents-sorter
+        files: '^\.Rbuildignore$'
     -   id: end-of-file-fixer
-    -   id: mixed-line-ending
-    -   id: trailing-whitespace
+        exclude: '\.Rd'
+-   repo: https://github.com/pre-commit-ci/pre-commit-ci-config
+    rev: v1.6.1
+    hooks:
+    # Only required when https://pre-commit.ci is used for config validation
+    -   id: check-pre-commit-ci-config
+-   repo: local
+    hooks:
+    -   id: forbid-to-commit
+        name: Don't commit common R artifacts
+        entry: Cannot commit .Rhistory, .RData, .Rds or .rds.
+        language: fail
+        files: '\.(Rhistory|RData|Rds|rds)$'
+        # `exclude: <regex>` to allow committing specific files
 #####
 # Python
```

Comment:
Do we even need the python stuff here? Or will there eventually be python in this repo?

Also, if we do want it, we can probably replace black and isort with ruff, since it can do all of what they do, and faster. See [here](https://github.com/cdcent/cfa-nssp-etl/blob/e44f300c51cb95bc519ebab62e64d82ea9c86e37/.pre-commit-config.yaml#L14) for an example. (Also, I personally don't like the line 79 character line length, and much prefer black's 88 char limit, if we up for changing anything).

Reply:
https://github.com/CDCgov/cfa-epinow2-pipeline/pull/2/commits/349363c0f19452d77aa1a2ed23da2b310a8cd219